machine/v1/types_nutanixprovider.go

```diff
@@ -0,0 +1,244 @@
+// +openshift:compatibility-gen:level=1
+// +k8s:openapi-gen=true
+type NutanixMachineProviderConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// clusterReferenceUuid is the UUID of the PE/cluster the Machine's VM will be created
+	ClusterReferenceUUID string `json:"clusterReferenceUuid,omitempty"`
+
+	// imageUuid is the UUID of the rhcos image uploaded to the PC.
+	// If the imageUUID is configured in the Machine CR, it will be used to create the VM.
+	// Otherwise, the imageName will be used to obtain the imageUUID, before creating the VM.
+	ImageUUID string `json:"imageUuid,omitempty"`
+
+	// imageName is the name of the rhcos image uploaded to the PC
+	ImageName string `json:"imageName,omitempty"`
+
+	// subnetUuid is the UUID of the network subnet to use for the Machine's VM
+	SubnetUUID string `json:"subnetUuid,omitempty"`
+
+	// numVcpusPerSocket is the number of vCPUs per socket of the VM to create
+	NumVcpusPerSocket int64 `json:"numVcpusPerSocket,omitempty"`
+
+	// numSockets is the number of sockets of the VM to create
+	NumSockets int64 `json:"numSockets,omitempty"`
+
+	// memorySizeMib is the memory size in megabytes of the VM to create
+	MemorySizeMib int64 `json:"memorySizeMib,omitempty"`
+
+	// diskSizeMib is the disk size in megabytes of the VM to create
+	DiskSizeMib int64 `json:"diskSizeMib,omitempty"`
+
+	// powerState is the expected power state of the VM to create
+	PowerState string `json:"powerState,omitempty"`
+
+	// userDataSecret is a local reference to a secret that contains the
+	// UserData to apply to the VM
+	UserDataSecret *corev1.LocalObjectReference `json:"userDataSecret,omitempty"`
+
+	// credentialsSecret is a local reference to a secret that contains the
+	// credentials data to access Nutanix PC client
+	CredentialsSecret *corev1.LocalObjectReference `json:"credentialsSecret,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NutanixMachineProviderConfigList contains a list of NutanixMachineProviderConfig
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+type NutanixMachineProviderConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []AlibabaCloudMachineProviderConfig `json:"items"`
+}
+
+// NutanixMachineProviderStatus is the type that will be embedded in a Machine.Status.ProviderStatus field.
+// It contains nutanix-specific status information.
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type NutanixMachineProviderStatus struct {
+	metav1.TypeMeta `json:",inline"`
```

Comment:
Do we need typemeta here?

Reply:
Not really, at least we don't use it in mapi-nutanix. I will remove it. But I see many other providers (alibaba, aws, etc.) use TypeMeta/ObjectMeta in their MachineProviderStatus struct.

Reply:
Ahh yes, actually, I think this might be needed. We embed the providerstatus within a rawextension, which I believe has a tag on it that checks for type meta within the resources, will double check that

Reply:
okay